### ISSUE TITLE
Whitelist support?

### ISSUE THREAD

**Issue by eddelbuettel** [MangoTheCat/rcmdcheck#12]
Checking of package isn't new or limited to R.  Debian has been doing it since time immortal as well with a tool called 'lintian'.  One _really_ useful feature is a whitelist one can add per-package.  Its format is really simple:  one item per row, and the item is the warning message.   So before counting / printing the warning, the override file is checked.

I think we should brainstorm how to support that. Maybe a file `inst/rccmdcheck.overrides.txt` ?  And in the output structure then have whitelist-adjusted counts?

**Comment by gaborcsardi** [comment]
Yes, this is one of the features that I wanted to implement, and it indeed needs some brainstorming.

Obviously I want to avoid CRAN submissions that check OK with `rcmdcheck`, but then are turned down by CRAN. So the whitelist should be specific, I guess, but I am not sure how we could enforce this, or if at all it is possible.

The R CMD check output varies a lot, and sometimes multiple checks are included in a single group. Ideally one should not whitelist a whole group, but only a specific check.

Anyway, I'll take a look at how lintian does it.

**Comment by gaborcsardi** [comment]
https://lintian.debian.org/manual/section-2.4.html is a good read and I think we can mostly follow that.

We can even start with a version without regular expressions, and your overrides having to match the check output **exactly**. Maybe with ignoring whitespace. E.g an entry could be like this:

```
# Yes, I know this package is big, but this is actually the "normal" size of it
checking installed package size ... NOTE
  installed size is 17.7Mb
  sub-directories of 1Mb or more:
    help 1.2Mb
    libs 14.8Mb
```

**Comment by eddelbuettel** [comment]
==:-)

The stoooopid size comment was also my personal bugbear of why we need that :)

How do you suggest dealing with one-line (in Lintian) versus multi-line (here) ?  Have 'paragraphs' in the override file, separate by empty lines?

**Comment by gaborcsardi** [comment]
Yeah, exactly, empty line as separator. If it turns out that there are empty lines within records, we can use some simple continuation mechanism, similarly to DCF.

**Comment by gaborcsardi** [comment]
Maybe we could just suppress particular NOTEs using env vars? What are the ones that we want to suppress? I could think of two:
- NOTE about the package size, for which we can set `_R_CHECK_PKG_SIZES_THRESHOLD_` to a higher value,
- NOTE about marked UTF-8 (etc.) string in data, this can be suppressed with `_R_CHECK_PACKAGE_DATASETS_SUPPRESS_NOTES_=true` I think.

Any others?
